src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -265,9 +266,15 @@ public function lists($column, $key = null)
      * @param  string  $pageName
      * @param  int|null  $page
      * @return \Illuminate\Contracts\Pagination\LengthAwarePaginator
+     *
+     * @throws InvalidArgumentException
      */
     public function paginate($perPage = null, $columns = ['*'], $pageName = 'page', $page = null)
     {
+        if ($perPage <= 0) {
+            throw new InvalidArgumentException("Negative values can't be used to query results");
```

Comment:
You mean RuntimeException?..I thought you said InvalidArgumentException was fine! and which other places are you talking about ?